router/throttler/adaptiveAlgorithmCounter/algorithm.go

```diff
@@ -0,0 +1,102 @@
+package adaptivethrottlercounter
+
+import (
+	"context"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+)
+
+type timer struct {
+	frequency    time.Duration
+	limitReached bool
+	mu           sync.Mutex
+	limitSet     bool
+	cancel       context.CancelFunc
```

Comment:
can you add a comment on what frequency, limitReached and limitSet is? 
What is the difference between limitReached and limitSet?

Reply:
Added comments